sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -450,9 +490,15 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
                 bindingDict["DataType"] = "Binary";
             }
 
+            var bindingNameAliasDict = GetPropertyNameAliasMapping(attribute);
```

Comment:
This will iterate through the properties twice, once to build the map and another to create the metadata. Can we delay the attribute verification (checking if the property name is overridden) for the loop below?

Reply:
Hm, The collection we are looping through below is a dictionary which contains the binding entry names and it's values.  The `GetAllDefinedProperties` using the [constructor param names](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/sdk/Sdk/Extensions/CustomAttributeExtensions.cs#L50) to build the binding (_with the assumption that it matches the property names_). I did not want to hook the override name check (which we can obtain from the the new attribute decoration of the properties only) to that because that code already is working on the naming assumption I mentioned above.  Yea, we have this additional iteration through the properties collection, but that seems more cleaner.

I just noticed our source gen version is also making this assumption (using constructor arguments). Let's see investigate we can switch to reading the property values.